api/v1beta1/condition_types.go

```diff
@@ -121,6 +121,7 @@ func SetTemporalWorkerProcessReady(w *TemporalWorkerProcess, status metav1.Condi
 		Message:            message,
 	}
 	apimeta.SetStatusCondition(&w.Status.Conditions, condition)
+	w.Status.Version = w.Spec.Version
```

Comment:
Could you please move this code to the reconcile function ? 
IMO it will be harder to maintain code that sets another field than `status.condition` in those functions types.

Reply:
yeah good call, ive moved it to reconcile loop which is better.